src/pack-objects.c

```diff
@@ -817,7 +832,9 @@ static int try_delta(git_packbuilder *pb, struct unpacked *trg,
 		trg_object->delta_data = NULL;
 	}
 	if (delta_cacheable(pb, src_size, trg_size, delta_size)) {
-		pb->delta_cache_size += delta_size;
+		if (git__add_uint64_overflow(&pb->delta_cache_size, pb->delta_cache_size, delta_size))
+			return -1;
+
 		git_packbuilder__cache_unlock(pb);
```

Comment:
If the value overflows, we'd end up leaving the cache locked, no? Should we unlock unconditionally and return -1 if it overflowed?